File: csharp/DVSClient/DVSClient/server/address/suggestions/RestApiSuggestionsFormatRequest.cs
using DVSClient.Address;
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Suggestions
{
    public class RestApiSuggestionsFormatRequest
    {
        [JsonProperty("country_iso")]
        public string? CountryIso { get; set; }

        [JsonProperty("datasets")]
        public IEnumerable<string>? Datasets { get; set; }

        [JsonProperty("max_suggestions")]
        public int? MaxSuggestions { get; set; }

        [JsonProperty("components")]
        public Address? Address { get; set; }

        [JsonProperty("layouts")]
        public IEnumerable<string>? Layouts { get; set; }

        public static RestApiSuggestionsFormatRequest Using(AddressConfiguration configuration)
        {
            var suggestionsFormat = new RestApiSuggestionsFormatRequest();

            // Country
            if (configuration.GetCountry() != null)
            {
                suggestionsFormat.CountryIso = configuration.GetCountry()?.Iso3Code;
            }

            // Datasets
            if (configuration.Datasets != null && configuration.Datasets.Any())
            {
                suggestionsFormat.Datasets = configuration.Datasets.ToList().ConvertAll(dataset => dataset.DatasetCode);
            }

            // Max suggestions
            if (configuration.MaxSuggestions != AddressConfiguration.DefaultMaxSuggestions)
            {
                suggestionsFormat.MaxSuggestions = configuration.MaxSuggestions;
            }

            // Layout name
            if (!string.IsNullOrEmpty(configuration.FormatLayoutName))
            {
                suggestionsFormat.Layouts = new List<string> { configuration.FormatLayoutName };
            }

            return suggestionsFormat;
        }
    }
}

File: typescript/src/server/address/suggestions/restApiSuggestionsFormatRequest.ts
import { AddressConfiguration } from "../../../address/addressConfiguration";
import { RestApiAddress } from "../restApiAddress";

export type RestApiSuggestionsFormatRequest = {
    country_iso?: string;
    datasets?: string[];
    max_suggestions?: number;
    components?: RestApiAddress;
    layouts?: string[];
};

export function getSuggestionsFormatRequestFromConfig(config: AddressConfiguration): RestApiSuggestionsFormatRequest {

    const request: RestApiSuggestionsFormatRequest = {};
    // Country
    if (config.options.datasets && config.options.datasets.length > 0) {
        request.country_iso = config.options.datasets[0].country.iso3Code;
    }
    // Datasets
    if (config.options.datasets && config.options.datasets.length > 0) {
        request.datasets = config.options.datasets.map(dataset => dataset.datasetCode);
    }
    // Max suggestions
    if (config.options.maxSuggestions !== AddressConfiguration.defaultMaxSuggestions) {
        request.max_suggestions = config.options.maxSuggestions;
    }
    //Layout name
    if (config.options.layoutName && config.options.layoutName.length > 0) {
        request.layouts = [config.options.layoutName];
    }

    return request;

};